src/smc_sagews/smc_sagews/tests/conftest.py

```diff
@@ -0,0 +1,394 @@
+        if md is not None  and len(md) > 0:
+            m['md'], did_truncate, tmsg  = t(md, sage_server.MAX_MD_SIZE, 'MAX_MD_SIZE')
+        if tex is not None and len(tex)>0:
+            tex['tex'], did_truncate, tmsg  = t(tex['tex'], sage_server.MAX_TEX_SIZE, 'MAX_TEX_SIZE')
+            m['tex'] = tex
+        if javascript is not None: m['javascript'] = javascript
+        if coffeescript is not None: m['coffeescript'] = coffeescript
+        if interact is not None: m['interact'] = interact
+        if d3 is not None: m['d3'] = d3
+        if obj is not None: m['obj'] = json.dumps(obj)
+        if file is not None: m['file'] = file    # = {'filename':..., 'uuid':...}
+        if raw_input is not None: m['raw_input'] = raw_input
+        if done is not None: m['done'] = done
+        if once is not None: m['once'] = once
+        if hide is not None: m['hide'] = hide
+        if show is not None: m['show'] = show
+        if events is not None: m['events'] = events
+        if clear is not None: m['clear'] = clear
+        if delete_last is not None: m['delete_last'] = delete_last
+        if did_truncate:
+            if 'stderr' in m:
+                m['stderr'] += '\n' + tmsg
+            else:
+                m['stderr'] = '\n' + tmsg
+        return m
+
+    def introspect_completions(self, id, completions, target):
+        m = self._new('introspect_completions', locals())
+        m['id'] = id
+        return m
+
+    def introspect_docstring(self, id, docstring, target):
+        m = self._new('introspect_docstring', locals())
+        m['id'] = id
+        return m
+
+    def introspect_source_code(self, id, source_code, target):
+        m = self._new('introspect_source_code', locals())
+        m['id'] = id
+        return m
+
+message = Message()
+
+###
+# end of copy region
+###
+
+def set_salvus_path(self, id):
+    r"""
+    create json message to set path and file at start of virtual worksheet
+    """
+    m = self._new('execute_code', locals())
+
+default_log_file = os.path.join(os.environ["HOME"], ".smc/sage_server/sage_server.log")
+
+def get_sage_server_info(log_file = default_log_file):
+    # log file ~/.smc/sage_server/sage_server.log
+    # sample sage_server startup line in first lines of log:
+    # 3136 (2016-08-18 15:02:49.372): Sage server 127.0.0.1:44483
+    try:
+        with open(log_file, "r") as inf:
+            for lno in range(5):
+                line = inf.readline().strip()
+                m = re.search("Sage server (?P<host>[\w.]+):(?P<port>\d+)$", line)
+                if m:
+                    host = m.group('host')
+                    port = int(m.group('port'))
+                    break
+            else:
+                raise ValueError('Server info not found in log_file',log_file)
+    except IOError:
+        pytest.fail("Unable to open log file %s\nYou may need to open a sage worksheet"%log_file)
```

Comment:
this and the previous error message are maybe a little bit too vague? Isn't in both cases the problem, that the sage server isn't running? I would rephrase that by something like: "there is probably no sage server running. you either have to open a sage worksheet or run `smc-sage-server start`" … at least, that's what I think needs to be done.